Лабораторная работа 3

Ivan16-cell opened the pull request.

AlexMTN commented on Лабораторная работа 3/Лаба 3 (Поиск общих участников.py:
```diff
@@ -0,0 +1,25 @@
+participants_first_group = "Иванов|Петров|Сидоров"
+participants_second_group = "Петров|Сидоров|Смирнов"
+
+common_participants = find_common_participants(participants_first_group, participants_second_group)
```
Хорошее решение!
Но в функцию вы подаете str с разделителем в виде |.
При вызове функции необходимо указать этот фактор.

AlexMTN commented on Лабораторная работа 3/Лаба 3 (Поиск общих участников.py:
```diff
@@ -0,0 +1,25 @@
+participants_first_group_comma = "Иванов,Петров,Сидоров"
+participants_second_group_comma = "Петров,Сидоров,Смирнов"
+
+print (find_common_participants(participants_first_group_comma, participants_second_group_comma, n=','))
```
Здесь дополнительно можно не назначать разделитель. Ваша функция изначально принимает за разделитель знак `,`

AlexMTN reviewed (changes requested):
Замечательно! Продолжайте в том же духе! 👍 
Обратите внимание на небольшой комментарий к заданию №2.

Ivan16-cell commented:
Здравствуйте. А почему на сайте в лабораторной номер 3 стоит крестик ? Я же все сдал и вы написали что все замечательно , как и предыдущих лабораторный , только с комментариями , что можно улучшить. Отправлено с iPhone8 дек. 2024 г., в 17:50, Aleksandr ***@***.***> написал(а):﻿
@AlexMTN requested changes on this pull request.

Замечательно! Продолжайте в том же духе! 👍
Обратите внимание на небольшой комментарий к заданию №2.

In Лабораторная работа 3/Лаба 3 (Поиск общих участников.py:
> +participants_first_group = "Иванов|Петров|Сидоров"
+participants_second_group = "Петров|Сидоров|Смирнов"
+
+# TODO Провеьте работу функции с разделителем отличным от запятой
+#print("Общие участники среди двух групп:", participants_first_group.intersection(participants_second_group))
+
+def find_common_participants(group1, group2, n=','):
+ first = set(group1.split(n))
+ second = set(group2.split(n))
+ common_participants = first.intersection(second)
+ return sorted(common_participants)
+
+participants_first_group = "Иванов|Петров|Сидоров"
+participants_second_group = "Петров|Сидоров|Смирнов"
+
+common_participants = find_common_participants(participants_first_group, participants_second_group)

Хорошее решение!
Но в функцию вы подаете str с разделителем в виде |.
При вызове функции необходимо указать этот фактор.



In Лабораторная работа 3/Лаба 3 (Поиск общих участников.py:
> +def find_common_participants(group1, group2, n=','):
+ first = set(group1.split(n))
+ second = set(group2.split(n))
+ common_participants = first.intersection(second)
+ return sorted(common_participants)
+
+participants_first_group = "Иванов|Петров|Сидоров"
+participants_second_group = "Петров|Сидоров|Смирнов"
+
+common_participants = find_common_participants(participants_first_group, participants_second_group)
+print(common_participants)
+
+participants_first_group_comma = "Иванов,Петров,Сидоров"
+participants_second_group_comma = "Петров,Сидоров,Смирнов"
+
+print (find_common_participants(participants_first_group_comma, participants_second_group_comma, n=','))

Здесь дополнительно можно не назначать разделитель. Ваша функция изначально принимает за разделитель знак ,

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

AlexMTN commented:
Добрый день!
Отправил работу на доработку, поскольку в строке 19 при вызове функции
`common_participants = find_common_participants(participants_first_group, participants_second_group)` 
Ваш код не выводит нужного результата.
Вы забыли явно указать параметр `n='|'` при вызове функции в 19 строке.

Ivan16-cell commented:
Добрый вечер! Спасибо за обратную связь . Понял в чем ошибка.С уважением, Таньков Иван.9 дек. 2024 г., в 11:07, Aleksandr ***@***.***> написал(а):﻿
Добрый день!
Отправил работу на доработку, поскольку в строке 19 при вызове функции
common_participants = find_common_participants(participants_first_group, participants_second_group)
Ваш код не выводит нужного результата.
Вы забыли явно указать параметр n='|' при вызове функции в 19 строке.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
